### PR TITLE
Make sure favicon, stores_icon and logo are correctly settled for themes

### LINKED ISSUE #21979: Logo, favicon and stores icon are not displayed in the 1.7.7.0-RC.1
#### Describe the bug

Since the latest changes on 1.7.7.x, themes are broken after an upgrade.

For example in the classic theme we now have:
```html
  <link rel="icon" type="image/vnd.microsoft.icon" href="{$urls.img_ps_url}{$shop.favicon}?{$shop.favicon_update_time}">
```
When it should be 
```html
  <link rel="icon" type="image/vnd.microsoft.icon" href="{$shop.favicon}?{$shop.favicon_update_time}">
```

#### Expected behavior

We should be able to reach `logo`, `stores_icon`, `favicon` without having to put the `_PS_IMG_` constant before.

#### Steps to Reproduce

Use a theme that is not the classic theme
For example: https://github.com/prestarocket-agence/classic-rocket

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: N/A

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | The `$urls.img_ps_url` should not be inserted in the template but in the controller to be sure themes are still working as expected..
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21979 
| How to test?  | Follow ticket instruction.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21981)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
PIng @okom3pom Thanks again for your reports :heart:

**Review by SimonGrn — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1608,10 +1611,12 @@ public function getTemplateVarCustomer($customer = null)
         return $cust;
     }
 
-    public function getTemplateVarShop()
+    public function getTemplateVarShop(array $urls = [])
```
Comment (eternoendless) [inline classes/controller/FrontController.php]: This is a BC for overrides 🤷‍♂️

File: themes/classic/templates/_partials/head.tpl [hunk 2]
```diff
@@ -51,8 +51,8 @@
 {/block}
 
 {block name='head_icons'}
-  <link rel="icon" type="image/vnd.microsoft.icon" href="{$urls.img_ps_url}{$shop.favicon}?{$shop.favicon_update_time}">
-  <link rel="shortcut icon" type="image/x-icon" href="{$urls.img_ps_url}{$shop.favicon}?{$shop.favicon_update_time}">
+  <link rel="icon" type="image/vnd.microsoft.icon" href="{$shop.favicon}?{$shop.favicon_update_time}">
```
Comment (eternoendless) [inline themes/classic/templates/_partials/head.tpl:54]: Won't this break themes using `{$urls.img_ps_url}{$shop.favicon}`?

Comment (PierreRambaud) [reply themes/classic/templates/_partials/head.tpl:54]: This is the original behaviour, it has been introduced in the 1.7.7.x and we decide to revert it.

**Review by eternoendless — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by ghost** [comment]
I cloned https://github.com/PierreRambaud/PrestaShop/tree/fix/21979 

`php tools/build/CreateRelease.php --version="1.7.7.0"`

i made an update without update default theme it's work 

https://cloud.ee.wtf/index.php/s/G99nsHTSBtnWMxY

**Review by ghost — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PierreRambaud,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1h0mkrpWQghZbVDlqhamC5bCRZSN3caC3/view?usp=sharing
Thank!

**Pull request merged**